pkg/api/v1/object-storage/buckets/bucket.go

```diff
@@ -85,18 +87,52 @@ func (svc *ObjectStorageBucketsService) Create(ctx context.Context, projectID, b
 	if err != nil {
 		return
 	}
-
 	_, err = svc.Client.Do(req, nil)
+	if err != nil {
+		return
+	}
+
+	w = wait.New(waitForCreation(ctx, projectID, bucketName, svc))
 	return
 }
 
+func waitForCreation(ctx context.Context, projectID string, bucketName string, svc *ObjectStorageBucketsService) wait.WaitFn {
```

Comment:
maybe convert it to a method of *ObjectStorageBucketsService instead of passing it as an attribute